clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -8048,11 +8048,16 @@ MipsABIInfo::classifyArgumentType(QualType Ty, uint64_t &Offset,
     // passing arguments according to the ABI
     unsigned Threshold = IsO32 ? 16 : 64;
     const TargetInfo &Target = getContext().getTargetInfo();
+    bool PassIndirect = false;
     if (Target.areAllPointersCapabilities()) {
       // assume we can pass structs up to 8 capabilities in size directly
       Threshold = 8 * (Target.getCHERICapabilityWidth() / 8);
+    } else if (const auto *RT = Ty->getAs<RecordType>()) {
+      // Aggregates containing capabilities are passed indirectly for hybrid
+      // varargs.
+      if (getContext().containsCapabilities(RT->getDecl()))
```

Comment:
Shouldn't this be only for the `!IsFixed` case? Otherwise I'd assume this breaks the current ABI.

Reply:
... yes, that's what I meant to write, and I assume is why a handful of tests are failing.